### ISSUE TITLE
Max channels in a zone

### ISSUE THREAD

**Issue by DustyLoon** [LibreDMR/OpenGD77_UserGuide#54]
A suggestion: There is a note in the GD-77S section that mentions the channel limits in a zone:

> Although the OpenGD77 CPS and codeplug format allows up to 80 channels per Zone, the GD-77S can only access the first 16 channels in each Zone, so codeplugs designed for the GD-77 containing more than 16 channels will need to be modified so that each zone only contains a maximum of 16 channels.

As I don't have a GD-77S, it didn't occur to me to look in that section for channel limitations. Perhaps a section that details the limitations of each model or the limitation could be mentioned in the CPS section (the note suggests it's a limitation of the CPS and not the radio).

Great job with the documentation! I find myself regularly referring to it and having "a-ha" moments all the time as I gain a better understanding of the firmware.

**Comment by rogerclarkmelbourne** [comment]
Its only the GD77S hardware which has this limitation because the radio its self uses a rotary switch to select the channel and it only has 16 steps.

All the other radios have UI that allows for all 80 channels.

Do you mean other limitations, other than channels ?

I'm not sure that we need to document that the GD77 etc don't have a GPS, as the owners should know this.
Also, owners should know that the GD77 has a black and white screen, so anything related to colour themes does not apply, and the themes section probably already point this out.
The RD5R hardware is so badly designed that hotspot mode doesnt work, but this is documented in the user guide already.

Pretty much everything else is the same across all supported radios

**Comment by DustyLoon** [comment]
You make a solid point, I should know what features my radio does and does not have. The docs would probably triple in size I suppose if you tried to document everything the radios do and don't have.

With the OpenGD firmware, I had no idea there was a 80 channel limitation for a zone for all radios (except the GD77S). Unless I missed it someplace, I only found it because I hit the 80 channel limit and wondered if I'd broken something or if this was by design. I didn't check the section about the GD77S previously as I don't have one, but since I was troubleshooting a specific issue, thanks to the ol' control-F I found the note about channel limitations in the GD77S section.

Hence my suggestion that the channel limitation should be notated in a place that all radio owners would read (perhaps with a note of the limitation of the GD77S, although having it covered in the GD77S section seems sufficient to me).

Thank you for the response!

**Comment by rogerclarkmelbourne** [comment]
The official GD77 firmware has a much lower channel limit per zone, I think it may be 32, I can't remember as we changed that ages ago by reducing the total number of zones so we could have more channels per zone, which resulted in the same total codeplug space being taken by the zones data. 

The codeplug format is fixed section.

I know some newer radios have codeplugs with dymanically resizing sections , but we can't do this without losing support for backward compatibility.

And as 80 channels per zone good enough for most applications there is no imperitive to change things


We can't ever document every aspect of every radio, its up to owners of the radios to do their own detailed research if they need to.

**Comment by DustyLoon** [comment]
I'm sorry if it appeared I was suggesting a change to the firmware - I thought this area was for reporting suggestions for the documentation. I was just thinking it would be nice to mention the 80 channel limit per zone someplace in the documentation that all firmware users would read it, instead of side note in a section for a specific radio model that people like me, perhaps in error, would skip since we don't have that radio.

Dynamically assigned channel counts to a zone would be cool, but then I'd rather have the compatibility of using the same code plug in two different radios I have here. The OpenGD firmware is just amazing.

**Comment by rogerclarkmelbourne** [comment]
I didn't think you were suggesting changing the codeplug limitations.

Probably the reason we never documented the entire codeplug, is because apart from the increase in channels per zone the codeplug limitations are the same as the official firmware.

We didn't design the codeplug format, its a legacy we inherited many years ago from supporting the official firmware.

We just added FM APRS support which is not in the official firmware, and this can only have 8 configurations. 
We will document this when the APRS functionality is stable, but as it was only released yesterday things are subject to change, including potentially having to reduce the maximum number of configs e.g. to 6, if more data is needed in each config, its too early to document it.
